### DISCUSSION TITLE
Usage with button on RSC

### DISCUSSION THREAD

**Discussion by ScreamZ** [nextauthjs/next-auth#10836] [Help] [answered]
The next JS documentation states that you can use `onClick` handler on RSC

https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions-and-mutations#event-handlers

But when using this code on RSC (no use client)

```ts
 onClick={async () => {
            "use server";
            await signIn();
          }}
```

I get:
> Only plain objects, and a few built-ins, can be passed to Server Actions. Classes or null prototypes are not supported.

If i do the same with a form action, this is OK

I know we have import { signIn } from "next-auth/react" that works, but this require client component.

ANy way to proceed ?

Thanks

**Comment by solo-samurai** [comment, marked as answer]
@ScreamZ they said 
> You can invoke a Server Action from event handlers such as onClick. 

### fix 1 RCC

```js
 "use server";
 
export const signInAction = async () => {
            await signIn();
}
```

```js
 "use client";

import { signInAction } from "actions.js"


<button  onClick={signInAction}></button>

 ```

### fix 2 RSC



```js

<button
        formAction={async () => {
          "use server";
          await signIn();
        }}
      ></button>
```
 

- you can't create actions in client components
- you can't use `onClick` on server components

**Reply by ScreamZ** [reply]
Damn you're good, thanks.

I'm getting lost as hell with all this syntax `use server` and this `formAction` property which is kind of dark magic. I guess I'll be used to over time.

**Reply by ScreamZ** [reply]
Still sounds terrifying 🤣 
<img width="1005" alt="Screenshot 2024-05-06 at 21 17 16" src="https://github.com/nextauthjs/next-auth/assets/6640835/8bb7176c-fe03-4d48-adc9-7bb074425389">
